### PR TITLE
Implemented Backend for Multiple Mentors functionality in a Group

### PR DESCRIPTION
**Pull request by aayush-05**
Additional Changes: Removed the "Update" end for Collaborations; since we are only either adding or removing a collaboration

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -9,11 +9,12 @@
   has_many :projects, foreign_key: "author_id", dependent: :destroy
   has_many :stars
   has_many :rated_projects, through: :stars, dependent: :destroy, source: "project"
-  has_many :groups_mentored, class_name: "Group",  foreign_key: "mentor_id", dependent: :destroy
+  has_many :groups_mentored, class_name: "Group", foreign_key: "primary_mentor_id",
+            dependent: :destroy
```
[the remark this answers is not in the document]
Comment (aayush-05) [reply app/models/user.rb:13]: Moving this to 4 separate lines isn't very apt I think

File: app/controllers/mentorships_controller.rb [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+class MentorshipsController < ApplicationController
+  before_action :set_mentorship, only: %i[destroy]
+
+  def self.policy_class
+    GroupPolicy
+  end
+
+  # POST /mentorships
+  # POST /mentorships.json
+  def create
```
[the remark this answers is not in the document]
Comment (aayush-05) [reply app/controllers/mentorships_controller.rb:12]: Any way out here? Shall I break it down to two methods?

File: app/controllers/mentorships_controller.rb [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    mentorship_emails = Utils.parse_mails_except_current_user(
+                        mentorship_params[:emails],
+                        current_user
+                        )
+
+    newly_added = mentorship_emails - already_present
+    newly_added.each do |email|
+      email = email.strip
+      user = User.find_by(email: email)
+      unless user.nil?
```
[the remark this answers is not in the document]
Comment (aayush-05) [reply app/controllers/mentorships_controller.rb:25]: Unless is fine too I guess

File: app/controllers/mentorships_controller.rb [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+class MentorshipsController < ApplicationController
+  before_action :set_mentorship, only: %i[destroy]
+
+  def self.policy_class
+    GroupPolicy
+  end
+
+  # POST /mentorships
+  # POST /mentorships.json
+  def create
```
[the remark this answers is not in the document]
Comment (aayush-05) [reply app/controllers/mentorships_controller.rb:12]: Any way out here?

**Comment by tachyons** [comment]
Fixed in https://github.com/CircuitVerse/CircuitVerse/commit/b9cf1f4ca5dada6b1977792ec49e40f44b15b3af

**Pull request closed without merging**